base_layer/wallet/src/transaction_service/storage/sqlite_db.rs

```diff
@@ -265,6 +265,24 @@ impl TransactionBackend for TransactionServiceSqliteDatabase {
             CompletedTransactionSql::find(tx_id, &(*conn)).is_ok())
     }
 
+    fn insert_completed_transaction(
```

Comment:
You don't need to write an explicit function to do this. Use the Write operation with the `DbKeyValuePair::CompletedTransaction(...,...)` It is already part of the unit tests etc.

Reply:
I agree a function does not need to be introduced but as the insert operation has to perform 3 steps, first acquire a lock, check if the key exists and then insert the completed transactions I believe this is more maintainable.

Reply:
Write handles the lock and if the key already exists will return a Duplicate error.

Reply:
Change applied, removed the method and made use of the write operation.